qiskit/converters/circuits_to_qobj.py

```diff
@@ -5,104 +5,145 @@
 # This source code is licensed under the Apache License, Version 2.0 found in
 # the LICENSE.txt file in the root directory of this source tree.
 
-"""Helper function for converting a list of circuits to a qobj"""
-from copy import deepcopy
+"""Compile function for converting a list of circuits to the qobj"""
 import uuid
+import warnings
 
-from qiskit.qobj import Qobj, QobjConfig, QobjExperiment, QobjItem, QobjHeader
 from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.qobj import Qobj, QobjConfig, QobjExperiment, QobjInstruction, QobjHeader
+from qiskit.qobj import QobjExperimentConfig, QobjExperimentHeader, QobjConditional
+from qiskit.qobj.run_config import RunConfig
 
 
-def circuits_to_qobj(circuits, backend_name, config=None, shots=1024,
-                     max_credits=10, qobj_id=None, basis_gates=None, coupling_map=None,
-                     seed=None, memory=False):
+def circuits_to_qobj(circuits, user_qobj_header=QobjHeader(), run_config=RunConfig(),
+                     qobj_id=None, backend_name=None,
+                     config=None, shots=None, max_credits=None,
+                     basis_gates=None,
+                     coupling_map=None, seed=None, memory=None):
     """Convert a list of circuits into a qobj.
 
     Args:
         circuits (list[QuantumCircuits] or QuantumCircuit): circuits to compile
-        backend_name (str): name of runner backend
-        config (dict): dictionary of parameters (e.g. noise) used by runner
-        shots (int): number of repetitions of each circuit, for sampling
-        max_credits (int): maximum credits to use
+        user_qobj_header (QobjHeader): header to pass to the results
+        run_config (RunConfig): RunConfig object
         qobj_id (int): identifier for the generated qobj
-        basis_gates (list[str])): basis gates for the experiment
-        coupling_map (list): coupling map (perhaps custom) to target in mapping
-        seed (int): random seed for simulators
-        memory (bool): if True, per-shot measurement bitstrings are returned as well
+
+        backend_name (str): to be deleted
+        config (dict): to be deleted
+        shots (int): to be deleted
+        max_credits (int): to be deleted
+        basis_gates (str): to be deleted
+        coupling_map (list): to be deleted
+        seed (int): to be deleted
+        memory (bool): to be deleted
```

Comment:
Can you add a note in the docstring (or even in a TODO somewhere) just to remember when are the deletions and deprecations to be completed (post-0.8?)? Mostly for not having to trace back the PR when the time for deprecation comes.

Reply:
Sure